Include <fcntl.h> to get loff_t definition

Patch borrowed from Alpine Linux at
http://git.alpinelinux.org/cgit/aports/plain/main/parted/fix-includes.patch. Solves
build with musl.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

--- parted-3.1.orig/libparted/fs/xfs/platform_defs.h
+++ parted-3.1/libparted/fs/xfs/platform_defs.h
@@ -35,6 +35,7 @@
 #define __XFS_PLATFORM_DEFS_H__
 
 #include <stdio.h>
+#include <fcntl.h>
 #include <stdarg.h>
 #include <assert.h>
 #include <endian.h>
